software/dimos/src/io/can.cpp

```diff
@@ -0,0 +1,62 @@
+#include "can.h"
+#include <cstring>
+#include <unistd.h>
+#include <net/if.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <linux/can.h>
+
+std::optional<int> CanInterface;
+
+Result<std::monostate, CanConnectionError> Can_Init() {
```

Comment:
Firmware coding convention is changing to `camelCase` after I'm done refactoring, which should be in a few weeks. This includes `jsoncan`. I don't care what you choose, just FYI if you were trying to stay in line with firmware.

Alternative suggestion: Use namespaces instead of module-prefixing since this is C++